core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -419,4 +422,33 @@ public void testCanReadOldCompressedManifestFiles() throws Exception {
     List<FileScanTask> tasks = Lists.newArrayList(reloaded.newScan().planFiles());
     Assert.assertEquals("Should scan 1 files", 1, tasks.size());
   }
+
+  @Test
+  public void testConcurrentAppend() throws Exception {
+    assertTrue("Should create v1 metadata",
+            version(1).exists() && version(1).isFile());
+    File dir = temp.newFolder();
+    dir.delete();
+
+    Table tableWithHighRetries = TABLES.create(SCHEMA, SPEC, new HashMap<String, String>() {
+      {
+        put(TableProperties.COMMIT_NUM_RETRIES, "1000");
+      }
+    }, dir.toURI().toString());
+    int threadsCount = 30;
```

Comment:
Nit / Somewhat blocking: Can we consider lowering this value? CI already takes some time, and I feel like spawning 30 threads in a test in our CI environment, where each of those threads runs a spark job, might be a little heavy given the very large number of concurrent users as well as the limited testing resources we get from github.

Would 4 or 5 threads be ok for all of the tests? Spawning 30 OS threads in one single test is likely to cause a large degree of contention.

What about if we changed threadsCount to 4 or 5? Would that still be ok?